secrets/awskms/kms.go

```diff
@@ -98,9 +99,28 @@ func (k *keeper) ErrorAs(err error, i interface{}) bool {
 }
 
 // ErrorCode implements driver.ErrorCode.
-func (k *keeper) ErrorCode(error) gcerrors.ErrorCode {
-	// TODO(shantuo): try to classify aws error codes
-	return gcerrors.Unknown
+func (k *keeper) ErrorCode(err error) gcerrors.ErrorCode {
+	ae, ok := err.(awserr.Error)
+	if !ok {
+		return gcerr.Unknown
+	}
+	ec, ok := errorCodeMap[ae.Code()]
+	if !ok {
+		return gcerr.Unknown
+	}
+	return ec
+}
+
+var errorCodeMap = map[string]gcerrors.ErrorCode{
+	kms.ErrCodeNotFoundException:          gcerrors.NotFound,
+	kms.ErrCodeDisabledException:          gcerrors.PermissionDenied,
+	kms.ErrCodeInvalidCiphertextException: gcerrors.InvalidArgument,
+	kms.ErrCodeKeyUnavailableException:    gcerrors.NotFound,
```

Comment:
Are you sure this means it's permanently gone, not just temporarily unavailable? If the latter, then we should probably use a different error.

Reply:
Not permanently gone, it can be retried. I'm not sure which one fits it properly. Maybe ResourceExhausted?

Reply:
Yes, exactly.